lib/index.js

```diff
@@ -105,26 +105,36 @@ const webAuth = (opener, opts, body) => {
 }
 
 const webAuthCheckLogin = (doneUrl, opts) => {
-  return fetch(doneUrl, opts).then(res => {
-    return Promise.all([res, res.json()])
-  }).then(([res, content]) => {
-    if (res.status === 200) {
-      if (!content.token) {
-        throw new WebLoginInvalidResponse('GET', res, content)
-      } else {
-        return content
-      }
-    } else if (res.status === 202) {
-      const retry = +res.headers.get('retry-after') * 1000
-      if (retry > 0) {
-        return sleep(retry).then(() => webAuthCheckLogin(doneUrl, opts))
+  const body = {
+    hostname: opts.hostname || os.hostname(),
+  }
+
+  return fetch(doneUrl, {
+    ...opts,
+    method: 'POST',
```

Comment:
@npm/cli: Since it's changing from a GET to a POST, is this then a breaking change?

Reply:
i'm also a little curious about using the hostname to determine "the same user" since really all it means is "a machine with the same name"

is the second request authenticated somehow? or are we relying on the likelihood of having the same hostname as another user and knowing the url to poll being minimal enough to not worry about it?

Reply:
if this is a security issue I don't know that we want the client defining this at all.

Reply:
agreed. common practice here is that the server generates a nonce when the initial login request is made and returns it as part of the polling url, that way you can be certain the requests looking for completion came from the same client that made the initial request

Reply:
This is signed off (and wanted) by security. 🙂 It's ensuring that the same metadata is present when the login request is initiated and when it is finished.

The hostname and IP are stored in the session here: https://cs.github.com/npm/registry-frontdoor/blob/d454b63a13e85f2d80717f872e3060cb7ae72275/routes/login.js#L18

Reply:
> that's a tricky one to answer.. it's been a GET for a long time, and i have no feel for how many folks out there may be using this functionality.
> 
> can we keep it a GET and put the hostname in the headers maybe?

@nlf: I changed the method back to `GET` and added a header instead